string field1
uint8 field2
bool field3
